core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -610,7 +611,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       s"${NETWORK_AUTH_ENABLED.key} must be enabled when enabling encryption.")
 
     val executorTimeoutThreshold = getTimeAsSeconds("spark.network.timeout", "120s")
-    val executorHeartbeatInterval = getTimeAsSeconds("spark.executor.heartbeatInterval", "10s")
+    val executorHeartbeatInterval = get(EXECUTOR_HEARTBEAT_INTERVAL).millis.toSeconds
```

Comment:
I think we've used TimeUnit directly for conversion mostly, rather than the Scala wrappers; I don't feel strongly about it, but it is what the rest of the file and related code uses.
`TimeUnit.SECONDS.convert(get(EXECUTOR_HEARTBEAT_INTERVAL), TimeUnit.MILLISECONDS)` 

But let's not introduce `EXECUTOR_HEARTBEAT_INTERVAL` in this PR; see below. This should be a narrow and simple change.